File: src/Application Specific/Classes/Entity Collections/Spares/PartTransaction/PartTransactionValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace PartTransactions
    {
        // make sure that contact object is valid
        public class PartTransactionValidator : BaseValidator
        {
            public void Validate(PartTransaction oPartTransaction)
            {

                // make sure that contact object is valid
                if (oPartTransaction.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oPartTransaction.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Spares/PartTransaction/PartTransactionValidator.vb
Namespace Entity

Namespace PartTransactions

Public Class PartTransactionValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oPartTransaction As PartTransaction)

        'make sure that contact object is valid
        If oPartTransaction.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oPartTransaction.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace
